### ISSUE TITLE
[Feature]: sp-combobox value is not equivalent to the currently selected menu item

### ISSUE THREAD

**Issue by nickschaap** [adobe/spectrum-web-components#4283]
### Code of conduct

- [X] I agree to follow this project's code of conduct.

### Impacted component(s)

Combobox

### Expected behavior

Let's say I have the following markup:
```html
<sp-combobox size="m" label="Color">
    <sp-menu-item value="red">Red</sp-menu-item>
    <sp-menu-item value="green">Green</sp-menu-item>
    <sp-menu-item value="blue">Blue</sp-menu-item>
</sp-combobox>
```
I would expect that when I select the "Red" menu item in the combobox that the `value` of the comboxbox changes to `red` which is the value of the selected menu item.

### Actual behavior

The `sp-comboxbox` value is `Red` which is the label of the menu item. Not the value.

### What browsers are you seeing the problem in?

Chrome

### How can we reproduce this issue?

1. Go to https://opensource.adobe.com/spectrum-web-components/components/combobox/#sizes
2. Open the comboxbox, click "Red"
3. Inspect the comboxbox element in the console
4. Check the `value` of the combobox
5. Observe that `value` as `Red`

**Comment by Westbrook** [comment]
The Combobox is less a fancy `<select>` (as seen with Picker) than a fancy `<input>` (as seen with Textfield) and should maintain a value that is the same as what is visible at all times. This can be particularly important when accepting values that are not represented by Menu Items provided to the Combobox, due to open user input. The conversion from value to key, as available, is pretty straight forward and we would definitely be open to a contribution of a property along the lines of `valueKey` (or an even classier name) to surface in parallel a matched item key when present.

**Comment by adixon-adobe** [comment]
I recently ran into this on our project, and we're finding it limiting that `sp-combobox` doesn't support the `value` attribute as this bug describes.

We'd like to take our hierarchical list of albums (each of which has an id) and represent it as a flat list in a combo box for selection. There can be multiple albums with the same name, so we need some way to differentiate these in the combo box.

There is an obvious conflict with user entered text that doesn't match any of the entries, so I understand the point @Westbrook is making, but it would be nice to have something like a `menu-value` property that tracks the menu value if present, or possibly have a mode where the value property is mapped to the menu item value rather than the text string.

We do need some way to handle multiple items with the same key, as the workaround are annoying and involve adding unwanted text to differentiate.
